src/MonoTorrent/MonoTorrent.Client/LocalPeerDiscovery.cs

```diff
@@ -152,6 +153,8 @@ async void ReceiveAsync (UdpClient client, CancellationToken token)
                     var uri = new Uri("ipv4://" + result.RemoteEndPoint.Address + ':' + portcheck);
 
                     PeerFound?.InvokeAsync (this, new LocalPeerFoundEventArgs (infoHash, uri));
+                } catch (FileNotFoundException ex) {
+                    throw ex;
```

Comment:
This should be `throw` so the full stacktrace is preserved, instead of `throw ex` which will lose the original stacktrace :) I'll tweak it and commit the changes!